usr/src/uts/common/fs/zfs/dsl_dataset.c

```diff
@@ -937,7 +965,7 @@ dsl_dataset_create_sync(dsl_dir_t *pdd, const char *lastname,
 		dsl_dataset_t *ds;
 
 		VERIFY0(dsl_dataset_hold_obj(dp, dsobj, FTAG, &ds));
-		dsl_dataset_zero_zil(ds, tx);
+		dsl_dataset_zero_zil(ds, tx, B_FALSE);
```

Comment:
To be honest, I have not properly investigated the above crash, so I can't really say why it happened.  One thing I recall that the problem was with a meta-dnode.